### PR TITLE
Add sensitivity to project based resources

### PR DESCRIPTION
**Pull request by CarsonF**
It's based on highest language engaged or project's own sensitivity for internships and calculated on read

Also moved role scoping & flattening into query

Part of #1736

### CODE REVIEW

**Comment by CarsonF** [comment]
Any thoughts on this @michaelmarshall @elisabeth-sorrell?

**Comment by michaelmarshall** [comment]
I'll take your word on your cypher updates to `matching.ts`:) We need have someone come behind and write some e2e tests on this.

**Review by elisabeth-sorrell — APPROVED** [review]
Slick! I guess the insecureRead isn't going into this iteration, yes?

File: src/components/project/project-member/project-member.service.ts [hunk 1]
```diff
@@ -197,24 +189,20 @@ export class ProjectMemberService {
...
         node('project', 'Project'),
         relation('out', '', 'member', { active: true }),
-        node('', 'ProjectMember', { id }),
+        node('node', 'ProjectMember', { id }),
+        relation('out', '', 'user'),
+        node('user', 'User'),
       ])
-      .with(['project', 'node', 'propList'])
-      .apply(matchMemberRoles(session.userId))
-      .match([node('node'), relation('out', '', 'user'), node('user', 'User')])
```
Comment (CarsonF) [inline src/components/project/project-member/project-member.service.ts:210]: @michaelmarshall Are you aware of performance implications of moving these 3 match statements into one? (as I have in new code).

**Comment by CarsonF** [comment]
> Slick! I guess the insecureRead isn't going into this iteration, yes?

Not this time 😄

File: src/components/project/project-member/project-member.service.ts (hunk 1 above)
Comment (michaelmarshall) [reply src/components/project/project-member/project-member.service.ts:210]: I do not. I've been surprised enough though. Not without testing on a large dataset would I be able to say.

**Comment by CarsonF** [comment]
I think I want to tackle moving the scopedRoles into the Resource classes here as well before this is merged.

**Comment by parashara1** [comment]
Successfully verified this PR: In order to verify that the sensitivity was implemented correctly, I tested the "ConsultantManager" role and its viewing capabilities regarding medium/low & high sensitivity projects. Created a Translation Project, assigned a universal template to the budget, create a managed & funded partnership with a date assigned of atleast one year. 
The outcome that was expected was that the sensitive fields in budget & budget records should appear blank to a "ConsultantManager" along with no presence of a Universal Template file even when one exists. Once you alter the sensitivity of a project and attach a language engagement with a Medium or Low sensitivity, the "ConsultantManager" should be able to see the universal template file and the budget/budget record data as well. 
The outcome that was expected occurred, and the "ConsultantManager" role seems to accurately have access to data corresponding to the proper sensitivities. 
Managed to test sensitivity on Internship Projects as well. Manually changed the sensitivity as a "ProjectManager" user and tested to see what viewing capabilities were affected for the "ConsultantManager" role. After discussing with Elisabeth, she said that this PR only has affected the "ConsultantManager" user with the Budget/Budget Record being the only attributes needing to be tested.  @ParkeBrown @michaelmarshall

**Comment by CarsonF** [comment]
I think I'd like to keep this PR just my changes to query and expose the sensitivity. Updating authorization to use this property is something entirely separate.

**Pull request merged**